Task-39609: Quarantine page: Display only the detected keywords in the document

AzmiTouil opened the pull request.

AzmiTouil committed:
Task-39609: Quarantine page: Display only the detected keywords in the document

rdenarie commented on core/connector/src/main/java/org/exoplatform/ecm/connector/dlp/FileDlpConnector.java:
```diff
@@ -148,8 +153,26 @@ private void saveDlpPositiveItem(Node node) throws RepositoryException {
     dlpPositiveItemService.addDlpPositiveItem(dlpPositiveItemEntity);
   }
+
+  private String getDetectedKeywords() {
```
Please add a test case to validate this function result

AzmiTouil replied:
Done !

AzmiTouil committed:
Task-39609: Add getDetectedKeywords UT

rdenarie commented on core/connector/src/main/java/org/exoplatform/ecm/connector/dlp/FileDlpConnector.java:
```diff
@@ -148,8 +153,26 @@ private void saveDlpPositiveItem(Node node) throws RepositoryException {
+        }
+      }
+    }
+    return StringUtils.join(excerptsList, ", ");
```
```suggestion
    return excerptsList.stream().collect(Collectors.joining(","));
```
I prefer to use join directly from java instead of using external lib. 
In addition you can remove the StringUtils import

rdenarie commented on core/connector/src/main/java/org/exoplatform/ecm/connector/dlp/FileDlpConnector.java:
```diff
@@ -148,8 +153,26 @@ private void saveDlpPositiveItem(Node node) throws RepositoryException {
+
+  private String getDetectedKeywords(Collection<SearchResult> searchResults) {
+    List<String> excerptsList = new ArrayList<>();
+    for (SearchResult searchResult : searchResults) {
+      Map<String, List<String>> excerpts = searchResult.getExcerpts();
+      for (List<String> excerptValue : excerpts.values()) {
+        if (excerptValue != null && !excerptValue.isEmpty()) {
+          Pattern pattern = Pattern.compile("<em>(.*?)</em>", Pattern.DOTALL);
+          Matcher matcher = pattern.matcher(excerptValue.get(0).toString());
+          while (matcher.find()) {
+            if (!excerptsList.contains(matcher.group(1))) {
+              excerptsList.add(matcher.group(1));
+            }
+          }
+        }
+      }
+    }
```
We should use a stream here

AzmiTouil committed:
PR Review

rdenarie committed:
Update function to not store searchResult as class object
Code should be stateless
Storing this searchResult as class parameter make it stateful

rdenarie committed:
Use stream for keyword detection

AzmiTouil committed:
Merge branch 'develop' into task-39609

AzmiTouil committed:
Task-39609: Display only the configured keywords

rdenarie commented on core/connector/src/main/java/org/exoplatform/ecm/connector/dlp/FileDlpConnector.java:
```diff
@@ -25,6 +25,7 @@
 import org.exoplatform.web.controller.metadata.ControllerDescriptor;
 import org.exoplatform.web.controller.router.Router;
 
+import java.lang.reflect.Array;
```
```suggestion
```

AzmiTouil committed:
remove useless import

rdenarie commented on core/connector/src/test/java/org/exoplatform/ecm/connector/dlp/TestFileDlpConnector.java:
```diff
@@ -160,6 +165,60 @@ public void testProcessItemWhenNotIndexed() throws Exception {
+    ArrayList<SearchResult> searchResults = new ArrayList<>();
+    searchResults.add(new SearchResult("url", "title", "excerpt", "detail", "imageUrl", 5, 4));
+
+    ArrayMap<String, List<String>> excerpts = new ArrayMap<>();
```
you should use java.util.Hashmap 
```suggestion
    Map<String, List<String>> excerpts = new HashMap<>();
```
Dont forget to remove the import

AzmiTouil committed:
Update UT
